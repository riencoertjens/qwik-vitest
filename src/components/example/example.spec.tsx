import { createDOM } from '@builder.io/qwik/testing';
import { test, expect } from 'vitest';
import {  VitestTestBroken } from './example';

test(`[VitestTestBroken Component]: Should render`, async () => {
  const { screen, render, userEvent } = await createDOM();
  await render(
    <VitestTestBroken />,
  );
  expect(screen.innerHTML).toMatchSnapshot();

  await userEvent('[data-test="button"]', 'click');
  expect(screen.innerHTML).toMatchSnapshot();
});
