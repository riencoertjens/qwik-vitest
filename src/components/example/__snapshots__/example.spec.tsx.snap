// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[VitestTestBroken Component]: Should render 1`] = `"<!--qv --><div><!--qv --><button data-test=\\"button\\" type=\\"button\\"><!--qv q:key q:sref=1 q:s-->hide button<!--/qv--></button><!--/qv--></div><!--/qv-->"`;

exports[`[VitestTestBroken Component]: Should render 2`] = `"<!--qv --><div><span>bbye</span><button data-test=\\"button\\" type=\\"button\\"><!--qv q:key q:sref=1 q:s-->hide button<!--/qv--></button><!--/qv--></div><!--/qv-->"`;
